dask/threaded.py

```diff
@@ -63,19 +64,19 @@ def get(dsk, result, cache=None, num_workers=None, pool=None, **kwargs):
         if pool is None:
             if num_workers is None and thread is main_thread:
                 if default_pool is None:
-                    default_pool = ThreadPool(CPU_COUNT)
-                    atexit.register(default_pool.close)
+                    default_pool = ThreadPoolExecutor(CPU_COUNT)
+                    atexit.register(default_pool.shutdown)
                 pool = default_pool
             elif thread in pools and num_workers in pools[thread]:
                 pool = pools[thread][num_workers]
             else:
-                pool = ThreadPool(num_workers)
-                atexit.register(pool.close)
+                pool = ThreadPoolExecutor(num_workers)
+                atexit.register(pool.shutdown)
                 pools[thread][num_workers] = pool
 
     results = get_async(
-        pool.apply_async,
-        len(pool._pool),
+        partial(executor_apply_async, pool),
+        pool._max_workers,
```

Comment:
In theory, [this should work]( https://github.com/python/cpython/blob/a041e116db5f1e78222cbf2c22aae96457372680/Lib/concurrent/futures/thread.py#L143 ). However it doesn't seem to work in practice.